### ISSUE TITLE
基础  全栈DS/DA养成手册

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#11399]
基础 — 全栈DS/DA养成手册<br>
<br>
https://ift.tt/5jtcXIw<br>
<br>
<br>
<br>

<div><div><ul>
<li>
<p>Epochs：一个epoch指的是用了trainset的所有数据进行了一轮训练</p>
<ul>
<li><p>The longer you train your model, the more tightly it will fit the training data (but may also overfit the test data).</p></li>
<li><p>To find the right number of epochs, monitor the loss on the validation data when training. Stop training when <strong>the validation loss begins increasing</strong> (rule of thumb).</p></li>
</ul>
</li>
<li>
<p>Batches：一次训练的时候只会用一个batch，# of batches越多越准确但更慢，有个speed-accuracy的</p>
<ul>
<li><p>Batch Gradient Descent (Full data)：只有一个batch（batch size就是length）</p></li>
<li>
<p><strong>Mini batch Gradient Descent</strong>：现在经常被误称为Batch Gradient Descent</p>

</li>
<li><p>Stochastic: batch size = 1</p></li>
</ul>
</li>
<li>
<p><strong>Batch size</strong>：The number of examples used per gradient update.</p>
<ul>
<li><p>A batch size of 1 is stochastic gradient descent. Many updates per epoch, but each is inaccurate.</p></li>
<li><p>A batch size of len(training_set) is batch gradient descent. One accurate update per epoch, but slow to compute.</p></li>
<li><p>Intermediate sizes (the default in Keras is 32) are called mini-batch gradient descent.</p></li>
</ul>
<p>In general, you will not need to change this parameter unless your examples are very small (structured data, in which case you can use a larger batch size).</p>
</li>
</ul></div></div><br>
<br>
<br>
<br>
<br>
<br>
via jace-yang.github.io https://ift.tt/rb8nUFS<br>
<br>
June 26, 2024 at 05:55PM
